XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs

```diff
@@ -338,38 +338,32 @@ private void GetXenServerVersions(XmlDocument xdoc)
 
         protected virtual XmlDocument FetchCheckForUpdatesXml()
         {
-            var xdoc = new XmlDocument();
+            var checkForUpdatesXml = new XmlDocument();
             var checkForUpdatesUrl = XenAdminConfigManager.Provider.GetCustomUpdatesXmlLocation() ?? BrandManager.UpdatesUrl;
-            var authToken = XenAdminConfigManager.Provider.GetInternalStageAuthToken();
-            var uri = new Uri(checkForUpdatesUrl);
+            var uriBuilder = new UriBuilder(checkForUpdatesUrl);
 
+            var uri = uriBuilder.Uri;
             if (uri.IsFile)
             {
-                xdoc.Load(checkForUpdatesUrl);
+                checkForUpdatesXml.Load(checkForUpdatesUrl);
             }
             else
             {
-                
+                var authToken = XenAdminConfigManager.Provider.GetInternalStageAuthToken();
+                uriBuilder.Query = Helpers.AddAuthTokenToQueryString(authToken, uriBuilder.Query);
+
                 var proxy = XenAdminConfigManager.Provider.GetProxyFromSettings(Connection, false);
 
                 using (var webClient = new WebClient())
                 {
                     webClient.Proxy = proxy;
                     webClient.Headers.Add("User-Agent", _userAgent);
-                    if (!string.IsNullOrEmpty(authToken))
-                    {
-                        NameValueCollection myQueryStringCollection = new NameValueCollection();
-                        myQueryStringCollection.Add(XenAdminConfigManager.Provider.GetInternalStageAuthTokenName(), authToken);
-                        webClient.QueryString = myQueryStringCollection;
-                    }
-
                     using (var stream = new MemoryStream(webClient.DownloadData(uri)))

```

Comment:
What value does uri have here? Isn't still the one from line 345? Don't we need the updated one including the query from line 353?